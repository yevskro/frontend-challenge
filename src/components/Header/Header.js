import React from 'react';
import styled from 'styled-components';

function Header(){
    return <header role='presentation'>
        <Image/>
    </header>
}

const Image = styled.img`
  height: 156px;
  background-color: #5DA5A4;
  @media (min-width: 375px) {
    content: url('/assets/images/bg-header-mobile.svg');
  }
  @media (min-width: 376px) {
    content: url('/assets/images/bg-header-desktop.svg');
  }
  @media (min-width: 1367px) { 
  /* 
      the svg is actually 1366px in width, most browsers go beyond that width
      and there will be white space on the right edge. this fills it
  */
    width: 100%;
  }
`;

export default Header;