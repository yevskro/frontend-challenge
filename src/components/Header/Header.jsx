import React from 'react';
import styled from 'styled-components';

function Header() {
  return (
    <header role="presentation">
      <Image />
    </header>
  );
}

const Image = styled.img`
  height: 156px;
  background-color: #5da5a4;
  background-image: url('/assets/images/bg-header-mobile.svg');
  width: 100%;
  @media (min-width: 376px) {
    background-image: url('/assets/images/bg-header-desktop.svg');
  }
`;

export default Header;
