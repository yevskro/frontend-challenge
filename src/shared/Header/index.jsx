import React from 'react';
import styled from 'styled-components';

function Header() {
  return (
    <header role="presentation">
      <Image src="/assets/images/blank.gif" alt="header banner" />
    </header>
  );
}

const Image = styled.img`
  height: 156px;
  background-color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
  background-image: url('/assets/images/bg-header-mobile.svg');
  width: 100%;
  @media (min-width: 376px) {
    background-image: url('/assets/images/bg-header-desktop.svg');
  }
`;

export default Header;
