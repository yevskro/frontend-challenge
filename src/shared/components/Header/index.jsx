import React from 'react';
import styled from 'styled-components';

function Header() {
  /* setting the image to blank.gif gets rid of the border that would be
  rendered if the src attribute is empty */
  return (
    <header role="presentation">
      <Image src="/assets/images/blank_image.gif" alt="header banner" />
    </header>
  );
}

const Image = styled.img`
  height: 156px;
  width: 100%;
  background-color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
  background-image: url('/assets/images/bg-header-mobile.svg');
  @media (min-width: 376px) {
    background-image: url('/assets/images/bg-header-desktop.svg');
  }
`;

export default Header;
