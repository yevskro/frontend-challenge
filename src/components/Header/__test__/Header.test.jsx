import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../Header';

test('renders without crashing', () => {
  const { getByRole } = render(<Header />);
  const appElement = getByRole('presentation');
  expect(appElement).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
