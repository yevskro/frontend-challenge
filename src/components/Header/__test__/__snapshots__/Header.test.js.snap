// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  height: 156px;
  background-color: #5DA5A4;
  content: url('/assets/images/bg-header-mobile.svg');
}

@media (min-width:376px) {
  .c0 {
    content: url('/assets/images/bg-header-desktop.svg');
  }
}

@media (min-width:1367px) {
  .c0 {
    width: 100%;
  }
}

<header
  role="presentation"
>
  <img
    className="c0"
  />
</header>
`;
